File: src/main/java/gg/mineral/practice/commands/spectator/SpectateCommand.java
package gg.mineral.practice.commands.spectator;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.entity.PlayerStatus;

import gg.mineral.practice.inventory.menus.SpectateMenu;
import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import lombok.val;

public class SpectateCommand extends PlayerCommand {

	public SpectateCommand() {
		super("spectate");
		setAliases("spec");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		val profile = ProfileManager.getOrCreateProfile(pl);

		if (profile.getPlayerStatus() != PlayerStatus.IDLE) {
			profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY);
			return;
		}

		if (args.length < 1) {
			profile.openMenu(new SpectateMenu());
			return;
		}

		val profileToSpectate = ProfileManager
				.getProfile(args[0],
						p -> p.getPlayerStatus() == PlayerStatus.FIGHTING || p.isInEvent());

		if (profileToSpectate == null) {
			profile.message(ErrorMessages.PLAYER_NOT_IN_MATCH_OR_EVENT);
			return;
		}

		profile.getSpectateHandler().spectate(profileToSpectate);
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/spectator/SpectateCommand.kt
package gg.mineral.practice.commands.spectator

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.inventory.menus.SpectateMenu
import gg.mineral.practice.util.messages.impl.ErrorMessages
import java.util.*

@Command(name = "spectate", aliases = ["spec"])
class SpectateCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg profileToSpectate: Optional<Profile>) {
        if (profile.playerStatus !== PlayerStatus.IDLE) return profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY)

        profileToSpectate.ifPresentOrElse({
            if (profile == it) profile.message(ErrorMessages.NOT_SPEC_SELF)
            else profile.spectate(it)
        }, { profile.openMenu(SpectateMenu()) })
    }
}
